### ISSUE TITLE
Keras framework issue

### ISSUE THREAD

**Issue by plagree** [bentoml/BentoML#2197]
Hello,

I'm currently trying out the pre-release of bentoml 1.0 and having issues with Keras models. When saving a Keras model through bentoml (bentoml.keras.save), I get the an error. The issue seems to be related to the h5 format which is used by bentoml under the hood when the recommended format (SavedModel) provides no error.

Bentoml version: `1.0.0a2`
Tensorflow version: `2.7.0`

Code to reproduce the error (from TF official documentation https://www.tensorflow.org/tutorials/structured_data/preprocessing_layers ).

```python
import bentoml
import numpy as np
import pandas as pd
import tensorflow as tf

from tensorflow.keras import layers

print(bentoml.__version__)
print(tf.__version__)

# Load data
dataset_url = 'http://storage.googleapis.com/download.tensorflow.org/data/petfinder-mini.zip'
csv_file = 'datasets/petfinder-mini/petfinder-mini.csv'
tf.keras.utils.get_file('petfinder_mini.zip', dataset_url, extract=True, cache_dir='.')
df = pd.read_csv(csv_file)
df.loc[:, 'target'] = np.where(df['AdoptionSpeed'] == 4, 0, 1)

def df_to_dataset(dataframe, shuffle=True, batch_size=32):
    df = dataframe.copy()
    labels = df.pop('target')
    df = {key: value[:,tf.newaxis] for key, value in dataframe.items()}
    ds = tf.data.Dataset.from_tensor_slices((dict(df), labels))
    if shuffle:
        ds = ds.shuffle(buffer_size=len(dataframe))
    ds = ds.batch(batch_size)
    ds = ds.prefetch(batch_size)
    return ds

def get_category_encoding_layer(name, dataset):
    index = layers.StringLookup(max_tokens=5)
    feature_ds = dataset.map(lambda x, y: x[name])
    index.adapt(feature_ds)
    encoder = layers.CategoryEncoding(num_tokens=index.vocabulary_size())
    return lambda feature: encoder(index(feature))

ds = df_to_dataset(df[['Type', 'target']])
my_input = tf.keras.Input(shape=(1,), name='Type', dtype='string')
encoding_layer = get_category_encoding_layer(name='Type', dataset=ds)
feature = encoding_layer(my_input)
x = tf.keras.layers.Dense(2, activation="relu")(feature)
output = tf.keras.layers.Dense(1)(x)
model = tf.keras.Model([my_input], output)
model.compile(optimizer='adam', loss=tf.keras.losses.BinaryCrossentropy(from_logits=True), metrics=["accuracy"])
model.fit(ds, epochs=2)
```

Then saving as a classic TF SavedModel provides not error
```python
model.save('my_pet_classifier')
```

But through BentoML 
```python
bentoml.keras.save('my_pet_classifier', model)
```
provides the following error

```bash
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
/tmp/ipykernel_699/2100602103.py in <module>
      1 # Save Keras model through BentoML
----> 2 bentoml.keras.save('my_pet_classifier', model)

/usr/local/lib/python3.7/site-packages/simple_di/__init__.py in _(*args, **kwargs)
    122         bind.apply_defaults()
    123 
--> 124         return func(*_inject_args(bind.args), **_inject_kwargs(bind.kwargs))
    125 
    126     setattr(_, "_is_injected", True)

/usr/local/lib/python3.7/site-packages/bentoml/_internal/frameworks/keras.py in save(name, model, store_as_json, custom_objects, metadata, model_store)
    188         model.save_weights(_model.path_of(_MODEL_WEIGHT_FNAME))
    189     else:
--> 190         model.save(_model.path_of(_SAVED_MODEL_FNAME))
    191 
    192     _model.save(model_store)

/usr/local/lib/python3.7/site-packages/keras/utils/traceback_utils.py in error_handler(*args, **kwargs)
     65     except Exception as e:  # pylint: disable=broad-except
     66       filtered_tb = _process_traceback_frames(e.__traceback__)
---> 67       raise e.with_traceback(filtered_tb) from None
     68     finally:
     69       del filtered_tb

/usr/local/lib/python3.7/site-packages/keras/saving/hdf5_format.py in _legacy_weights(layer)
    970   if any(not isinstance(w, tf.Variable) for w in weights):
    971     raise NotImplementedError(
--> 972         f'Save or restore weights that is not an instance of `tf.Variable` is '
    973         f'not supported in h5, use `save_format=\'tf\'` instead. Received a '
    974         f'model or layer {layer.__class__.__name__} with weights {weights}')

NotImplementedError: Save or restore weights that is not an instance of `tf.Variable` is not supported in h5, use `save_format='tf'` instead. Received a model or layer StringLookup with weights [<keras.layers.preprocessing.index_lookup.VocabWeightHandler object at 0x7fa47830a6d0>]
```

**Comment by larme** [comment]
Thank you for reporting the issue.

Currently `bentoml.keras.save` use h5 as internal format because we want to keep compatibility with older version of tensorflow. We will add an option to save with SavedModel format later. Meanwhile you may try to use `bentoml.tensorflow.save` to save your Keras model as a temporary solution.

**Comment by aarnphm** [comment]
fixed with #2329. Feel free to reopen if you still have problem @plagree
